lib/api/core/models/security/role.js

```diff
@@ -352,3 +345,193 @@ Role.prototype.validateDefinition = function (context) {
+    .then(args => {
+      var sandboxContextObject = {
+        $requestObject: requestObject,
+        context: context,
+        $currentUserId: context.token.user._id,
+        args: args
+      };
+
+      sandboxContext = vm.createContext(sandboxContextObject);
+
+      if (!this.closures[path].test) {
+        try {
+          sandboxScript = new vm.Script('(function ($requestObject, context, $currentUserId, args) { ' +
+            actionRights.test +
+            '\nreturn false;\n })($requestObject, context, $currentUserId, args)');
+
+          this.closures[path].test = sandboxScript;
+        }
+        catch (err) {
+          message = `Error parsing closure rights for role ${this._id} (${path.join('/')}): ${actionRights}`;
+
+          kuzzle.pluginsManager.trigger('log:error', message);
+          error = new InternalError(message);
+          error.details = err;
+
+          throw (error);
+        }
+      }
+
+      return this.closures[path].test.runInContext(sandboxContext);
+    })
+    .catch((err) => {
+      message = `Error during executing rights action closure (${path.join('/')}): ${err.message}`;
+
+      kuzzle.pluginsManager.trigger('log:error', message);
+      error = new InternalError(message);
+      error.details = err;
+
+      throw (error);
+    });
+}
+
+/**
+ *
+ * @param argsDefinitions
+ * @returns {Promise}
+ */
+function buildArgsForContext (argsDefinitions) {
+  if (!_.isEmpty(argsDefinitions)) {
+    return buildClosureArgs.call(this, argsDefinitions);
+  }
+
+  return q({});
+}
+
+/**
+ * @param {Object} argsDefinitions
+ * @returns {Promise}
+ */
+function buildClosureArgs (argsDefinitions) {
+  var
+    deferred = q.defer(),
+    argsFunctions = {};
+
+  // Build the object that will be passed to parallel
+  _.forEach(argsDefinitions, (argDefinition, argName) => {
+    argsFunctions[argName] = factoryFunctionClosure.call(this, argName, argDefinition);
+  });
+
+  async.parallel(argsFunctions, (error, results) => {
+    if (error) {
+      // In case we have an error we want to return an empty object because the error can come from readEngine
+      return deferred.resolve({});
+    }
+
+    deferred.resolve(results);
+  });
+
+  return deferred.promise;
+}
+
+/**
+ * Return a function built for async.parallel with the right action on readEngine
+ * @param {String} argName - the argument name set by user (i.e.: document, documents...)
+ * @param {Object} argDefinition - definition set by user. Like {collection: 'messages', index: 'chat', action: { get: '$currentId' }}
+ * @returns {Function} the function that will execute a get/mget/search on readEngine
+ */
+function factoryFunctionClosure (argName, argDefinition) {
+  var methodName;
+
+  if (!argDefinition.collection || !argDefinition.index || !argDefinition.action || _.isEmpty(argDefinition.action)) {
+    this.pluginsManager.trigger('log:error', `Bad format in closure rights for ${argName}`);
+    return callback => callback(null, {});
```

Comment:
Why are we returning a silent failure?
Wouldn't it be better to return `callback => callback(new BadRequestError(...))`?